queries/sway/highlights.scm

```diff
@@ -0,0 +1,336 @@
+
+(use_as_clause
+  path: (identifier)? @module
+  alias: (identifier) @module)
+
+; ---
+; Remaining Paths
+; ---
+(scoped_identifier
+  path: (identifier)? @module
+  name: (identifier) @module)
+
+(scoped_type_identifier
+  path: (identifier) @module)
+
+[
+  "*"
+  "'"
+  "->"
+  "=>"
+  "<="
+  "="
+  "=="
+  "!"
+  "!="
+  "%"
+  "%="
+  "&"
+  "&="
+  "&&"
+  "|"
+  "|="
+  "||"
+  "^"
+  "^="
+  "*"
+  "*="
+  "-"
+  "-="
+  "+"
+  "+="
+  "/"
+  "/="
+  ">"
+  "<"
+  ">="
+  ">>"
+  "<<"
+  ">>="
+  "<<="
+  "@"
+  ".."
+  "..="
+  "'"
+  "?"
+] @operator
+
+(use_wildcard
+  "*" @character.special)
+
+[
+  (string_literal)
+  (raw_string_literal)
+] @string
+
+[
+  (line_comment)
+  (block_comment)
+] @comment
+
+; ---
+; Extraneous
+; ---
+(self) @variable.builtin
+
+(enum_variant
+  (identifier) @constant)
+
+(field_initializer
+  (field_identifier) @variable.member)
+
+(shorthand_field_initializer
+  (identifier) @variable.member)
+
+(shorthand_field_identifier) @variable.member
+
+(loop_label
+  "'" @label
+  (identifier) @label)
+
+; ---
+; Punctuation
+; ---
+[
+  "::"
+  ":"
+  "."
+  ";"
+  ","
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+  "#"
+] @punctuation.bracket
+
+(type_arguments
+  [
+    "<"
+    ">"
+  ] @punctuation.bracket)
+
+(type_parameters
+  [
+    "<"
+    ">"
+  ] @punctuation.bracket)
+
+(closure_parameters
+  "|" @punctuation.bracket)
+
+(let_declaration
+  pattern: [
+    (identifier) @variable
+    (tuple_pattern
+      (identifier) @variable)
+  ])
+
+; It needs to be anonymous to not conflict with `call_expression` further below.
+(_
+  value: (field_expression
+    value: (identifier)? @variable
+    field: (field_identifier) @variable.member))
+
+(parameter
+  pattern: (identifier) @variable.parameter)
+
+(parameter
+  pattern: (ref_pattern
+    [
+      (mut_pattern
+        (identifier) @variable.parameter)
+      (identifier) @variable.parameter
+    ]))
+
+(closure_parameters
+  (identifier) @variable.parameter)
+
+(for_expression
+  "for" @keyword.repeat)
+
+"in" @keyword.repeat
+
+[
+  "match"
+  "if"
+  "else"
+] @keyword.conditional
+
+"while" @keyword.repeat
+
+[
+  "break"
+  "continue"
+  "return"
+  "yield"
+] @keyword.return
+
+"use" @keyword.import
+
+(mod_item
+  "mod" @keyword.import
+  !body)
+
+(use_as_clause
+  "as" @keyword.import)
+
+(type_cast_expression
+  "as" @keyword.operator)
+
+[
+  "as"
+  "mod"
+  "abi"
+  "impl"
+  "where"
+  "trait"
+  "for"
+  "let"
+  "contract"
+  "script"
+  "predicate"
+  "library"
+] @keyword
+
+[
+  "struct"
+  "enum"
+  "storage"
+  "configurable"
+  "type"
+] @keyword.type
+
+[
+  "fn"
+  "abi"
+] @keyword.function
+
+[
+  (mutable_specifier)
+  "const"
+  "ref"
+  "deref"
+  "move"
+  "pub"
+] @keyword.modifier
+
+(reference_type
+  "&" @keyword.modifier)
+
+(self_parameter
+  "&" @keyword.modifier)
+
+; -------
+; Guess Other Types
+; -------
+((identifier) @constant
+  (#lua-match? @constant "^[A-Z][A-Z%d_]*$"))
```

Comment:
As talked about before, I _really_ want to do away with these "convention captures" -- with tree-sitter, we should be able to do better.

This is not specific to this language, of course, but I'm wondering if at some point we should stop adding new ones even before removing the old ones?

Reply:
And obviously not a blocker; just food for thought.

Reply:
Sure, that makes sense. Personally I think in that case we should stop allowing these queries for new languages, although maybe this could be a special case since the language is so similar to rust. Maybe would make more sense to remove them here at the time of removing the rust ones (which maybe would be when removing all of them?)

Reply:
Sure, as I said, don't want to block this!